src/core/ddsi/src/q_entity.c

```diff
@@ -620,7 +725,22 @@ dds_return_t new_participant_guid (const ddsi_guid_t *ppguid, struct q_globals *
       if (ret != DDS_RETCODE_OK)
         goto new_pp_err_secprop;
     }
+
+    if (!q_omg_security_check_create_participant (pp, gv->config.domainId))
```

Comment:
I don't see why this check needs to be duplicated. If the ``pp->plist->qos`` doesn't specify a security config and there is a ``gv->config.omg_security_configuration`` it copies the latter in ``pp->plist->qos``. After that step, you are in the same state and can do with a single check.